checker/src/org/checkerframework/checker/index/lowerbound/LowerBoundAnnotatedTypeFactory.java

```diff
@@ -95,6 +95,42 @@ public LowerBoundAnnotatedTypeFactory(BaseTypeChecker checker) {
                         LowerBoundUnknown.class));
     }
 
+    /**
+     * Takes a value type (only interesting if it's an IntVal), and converts it to a lower bound
+     * type. If the new lower bound type is more specific than type, convert type to that type.
+     *
+     * @param valueType the value checker's type
+     * @param type the current lower bound type of the expression being evaluated
+     */
+    private void addLowerBoundTypeFromValueType(
+            AnnotatedTypeMirror valueType, AnnotatedTypeMirror type) {
+        AnnotationMirror anm = getLowerBoundAnnotationFromValueType(valueType);
+        if (!type.isAnnotatedInHierarchy(UNKNOWN)) {
+            type.addAnnotation(anm);
```

Comment:
Thanks. I don't really understand what you mean by the "primary" annotation, but it's likely an implementation detail of the framework I just haven't encountered before. The solution is quite concrete.